dask_ml/model_selection/_hyperband.py

```diff
@@ -0,0 +1,532 @@
+
+    References
+    ----------
+    .. [1] "Hyperband: A novel bandit-based approach to hyperparameter
+           optimization", 2016 by L. Li, K. Jamieson, G. DeSalvo, A.
+           Rostamizadeh, and A. Talwalkar.  https://arxiv.org/abs/1603.06560
+
+    """
+
+    def __init__(
+        self,
+        estimator,
+        parameters,
+        max_iter=81,
+        aggressiveness=3,
+        test_size=None,
+        patience=False,
+        tol=1e-3,
+        random_state=None,
+        scoring=None,
+    ):
+        self.max_iter = max_iter
+        self.aggressiveness = aggressiveness
+
+        super(HyperbandSearchCV, self).__init__(
+            estimator,
+            parameters,
+            max_iter=max_iter,
+            patience=patience,
+            tol=tol,
+            test_size=test_size,
+            random_state=random_state,
+            scoring=scoring,
+        )
+
+    def _get_SHAs(self, brackets):
+
+        patience = _get_patience(self.patience, self.max_iter, self.aggressiveness)
+
+        # This is the first time self.random_state is used after
+        # HyperbandSearchCV.fit is called.
+        seed_start = check_random_state(self.random_state).randint(2 ** 31)
+        self._SHA_seed = seed_start
+
+        # These brackets are ordered by adaptivity; bracket=0 is least adaptive
+        SHAs = {
+            b: SuccessiveHalvingSearchCV(
+                self.estimator,
+                self.parameters,
+                n_initial_parameters=n,
+                n_initial_iter=r,
+                aggressiveness=self.aggressiveness,
+                max_iter=self.max_iter,
+                patience=patience,
+                tol=self.tol,
+                test_size=self.test_size,
+                random_state=seed_start + b if b != 0 else self.random_state,
+                scoring=self.scoring,
+            )
+            for b, (n, r) in brackets.items()
+        }
+        return SHAs
+
+    @gen.coroutine
+    def _fit(self, X, y, **fit_params):
+        X = self._check_array(X)
+        y = self._check_array(y, ensure_2d=False)
+        scorer = check_scoring(self.estimator, scoring=self.scoring)
+
+        if self.max_iter < 1:
+            raise ValueError("max_iter < 1 is not supported")
+
+        brackets = _get_hyperband_params(self.max_iter, eta=self.aggressiveness)
+
+        SHAs = self._get_SHAs(brackets)
+        # Which bracket to run first? Going to go with most adaptive;
+        # hopefully less adaptive can fill in for any blank spots
+        #
+        # _brackets_ids is ordered from largest to smallest
+        _brackets_ids = list(reversed(sorted(SHAs)))
+        _SHAs = yield [SHAs[b]._fit(X, y, **fit_params) for b in _brackets_ids]
+        SHAs = {b: SHA for b, SHA in zip(_brackets_ids, _SHAs)}
+
+        # This for-loop rename estimator IDs and pulls out wall times
+        key = lambda b, old: "bracket={}-{}".format(b, old)
```

Comment:
Style nit: the `lambda` is unnecessary here. You can just do `key = "bracket={}-{}".format`.

Reply:
D'oh